### PR TITLE
Add shoryuken async strategy

### PR DESCRIPTION
**Pull request by josephchoe**
I added a new strategy for AWS SQS-based queues using the shoryuken gem.

### CODE REVIEW

**Comment by pyromaniac** [comment]
Cool, can you also fix the readme and the changelog please?

**Comment by josephchoe** [comment]
Done.

**Comment by pyromaniac** [comment]
ok, can you add the changelog entry on top, please? Also, did you test it manually? Does it work? If so, I'll merge it as soon as specs pass

**Comment by josephchoe** [comment]
Oops, sorry.

**Comment by josephchoe** [comment]
The specs pass, and I've tested it manually. But I think I've noticed a problem in copying the worker and specs from the sidekiq strategy to shoryuken. Shoryuken only passes in sqs_msg and body into the worker and not options, which would be a problem for passing in `suffix`.

https://github.com/phstc/shoryuken/blob/master/lib/shoryuken/processor.rb#L17

**Comment by pyromaniac** [comment]
can we use `Shoryuken::Worker.perform_async({index: type, ids: ids, options: options}, queue: shoryuken_queue)` ?

File: lib/chewy/strategy/shoryuken.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        def perform(_sqs_msg, body, options = {})
+          options[:refresh] = !Chewy.disable_refresh_async if Chewy.disable_refresh_async
+          body['index'].constantize.import!(body['ids'], options)
+        end
+      end
+
+      def leave
+        @stash.each do |type, ids|
+          next if ids.empty?
+          Shoryuken::Worker.perform_async({index: type, ids: ids}, queue: shoryuken_queue)
```
Comment (pyromaniac) [inline lib/chewy/strategy/shoryuken.rb]: Also, how about to use `{type: type` ?

Comment (josephchoe) [inline lib/chewy/strategy/shoryuken.rb]: Will do.

**Comment by josephchoe** [comment]
Made the requested changes. Let me know if there's anything else that needs to be addressed.

**Comment by pyromaniac** [comment]
Cool, merged, thanks!

**Pull request merged**